apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -105,8 +37,10 @@ export const VariableSelect = ({
   const [filterValue, setFilterValue] = useState('');
   const [isOpen, setIsOpen] = useState(false);
   const [options, setOptions] = useState(optionsProp);
-  const [hoveredOptionIndex, setHoveredOptionIndex] = useState(0);
-  const variablesListRef = useRef<HTMLUListElement>(null);
+
+  const { variablesListRef, hoveredOptionIndex, next, prev, reset, focusFirst } = useListKeyboardNavigation({
```

Comment:
TODO: better name

Reply:
Do we need a a separate hook for this? Since variables-list is a reusable component I'd expect the hook to live in there.

Reply:
yes we need it as it allows to control the list from the parent... for example `MailyVariablesList` has the `useImperativeHandle` with `onKeyDown` which is called by it's parent